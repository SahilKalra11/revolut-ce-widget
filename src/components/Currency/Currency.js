import React, {Component} from 'react';
import './Currency.scss';

class Currency extends Component {
    render() {
        return (
            <section className="exchange__currency">
                <div className="exchange__currency-name">
                    {this.props.currency}
                </div>
            </section>
        );
    }
}

export default Currency;
