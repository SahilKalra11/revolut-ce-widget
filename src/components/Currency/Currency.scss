@import '../../utilities/variables';
@import '../../utilities/mixins';

.exchange__currency-name {
  @include currency-style;
}

li .exchange__currency-name {
  color: $color-blue;
  font-size: 2rem;
}
