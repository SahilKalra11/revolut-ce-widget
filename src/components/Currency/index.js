export {default} from './Currency.js';
