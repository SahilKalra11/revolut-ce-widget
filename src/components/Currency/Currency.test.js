import React from 'react';
import ReactDOM from 'react-dom';
import render from 'react-test-renderer';
import Currency from './Currency';
import {unmountComponentAtNode} from "react-dom";


describe('Currency', () => {
    let container = null;
    beforeEach(() => {
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    const props = {
        currency: 'USD',
    };

    it('renders component without crashing', () => {
        ReactDOM.render(
            <Currency {...props} />,
            container
        );
    });

    test('matches to snapshot', () => {
        const component = render.create(
            <Currency {...props} />
        );
        let componentTree = component.toJSON();
        expect(componentTree).toMatchSnapshot();
    });
});
