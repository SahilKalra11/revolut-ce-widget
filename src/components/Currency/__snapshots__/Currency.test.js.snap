// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency matches to snapshot 1`] = `
<section
  className="exchange__currency"
>
  <div
    className="exchange__currency-name"
  >
    USD
  </div>
</section>
`;
